### PR TITLE
Decorator

### PR DESCRIPTION
**Pull request by DyaryRaoof**
- Created a class Corrector with a method correct_name that does the following:
- Made sure the first letter of the given name is a capital letter.
- Made sure that the name has a maximum of 10 characters. If it's longer it should trim the word.
- Set an instance of Corrector in Person on initialization.
- In Person added a method validate_name that uses the Corrector instance to validate and save the corrected name.

### CODE REVIEW

File: person.rb [hunk 1]
```diff
@@ -21,5 +24,9 @@ def can_use_services?
     end
   end
 
+  def validate_name
+    @name = @corrector.correct_name
```
Comment (GzimAsani) [inline person.rb]: - [ ] Please update this line to:

`@name = @corrector.correct_name(@name)`

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi @DyaryRaoof                              
Good job so far! Your code looks fine, good job :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Great job with the project it looks really good :muscle: 

## Required Changes ♻️ 
- [ ] _Check inline comments for more information :arrow_double_down:_ 


## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: person.rb (hunk 1 above)
Comment (DyaryRaoof) [reply person.rb]: OK. though i am taking the name from the the class constructor so no need to pass in  the name there.

Comment (GzimAsani) [reply person.rb]: - [ ] You're right I missed that

**Comment by helman101** [comment]
# STATUS: APPROVED 🟢 🥳 

Hi @DyaryRaoof 🥇 , 

### Congratulations!!! 🥳 🎉

You have successfully completed all the expectations for this project, there is nothing more to do here, now you can go on your way and keep improving your skills.

Cheers and Happy coding!👏👏👏

gif

If anything is not 100% clear please leave your questions or comments in the PR thread, we will be glad to help you 😄.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**